test/event/handlers/RaidDetectionHandlerTest.ts

```diff
@@ -0,0 +1,64 @@
+import { expect } from "chai";
+import { Constants } from "discord.js";
+import RaidDetectionHandler from "../../../src/event/handlers/RaidDetectionHandler";
+import { SinonSandbox, createSandbox } from "sinon";
+import MockDiscord from "../../MockDiscord";
+import { RAID_SETTINGS, MODS_CHANNEL_ID } from "../../../src/config.json";
+
+describe("RaidDetectionHandler", () => {
+	describe("constructor()", () => {
+		it("creates a handler for GUILD_MEMBER_ADD", () => {
+			const handler = new RaidDetectionHandler();
+
+			expect(handler.getEvent()).to.equal(Constants.Events.GUILD_MEMBER_ADD);
+		});
+	});
+
+	describe("handle()", () => {
+		let sandbox: SinonSandbox;
+		let handler: RaidDetectionHandler;
+		let discordMock: MockDiscord;
+
+		beforeEach(() => {
+			sandbox = createSandbox();
+			handler = new RaidDetectionHandler();
+			discordMock = new MockDiscord();
+		});
+
+		it("adds a member to the joinQueue", async () => {
+			const mockGuildMember = discordMock.getGuildMember();
+
+			await handler.handle(mockGuildMember);
+
+			expect(handler.joinQueue.includes(mockGuildMember)).to.be.true;
+		});
+
+		it("removes member from joinQueue", done => {
+			const mockGuildMember = discordMock.getGuildMember();
+
+			handler.handle(mockGuildMember).then(() => {
+				expect(handler.joinQueue.includes(mockGuildMember)).to.be.true;
+				setTimeout(() => {
+					expect(handler.joinQueue.includes(mockGuildMember)).to.be.false;
+					done();
+				}, 1000 + RAID_SETTINGS.TIME_TILL_REMOVAL * 1000);
+			});
```

Comment:
Is there a reason you are using `.then()` over `async/await` here?

Reply:
that was the only way to make it work. see discord bot channel